daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/ConverterOps2.scala

```diff
@@ -78,3 +78,15 @@ case object StringToTime extends Converter {
     result
   }
 }
+
+case object StringToHexBinary extends Converter with HexBinaryKind{
+  val name = "StringToHexBinary"
+
+  override def computeValue(a: AnyRef, dstate: DState): AnyRef = {
+    val result = a match {
+      case s: String => hexStringToByteArray(s)
+      case x => throw new NumberFormatException("%s cannot be cast to dfdl:hexBinary\ndfdl:hexBinary received an unrecognized type! Must be String or HexBinary.".format(x.toString))
+    }
+    result
+  }
+}
```

Comment:
Does it make sense to modify XSHexBinary to call this function now so as to minimize duplicate code? So XSHexBInary basically just calls StringToHexBinary.computeValue(...), and all the actual conversion logic is only in this object?  This looks like how we do other conversions like XSTime vs StringToTime

Reply:
I wasn't sure that it would, since the XSHexBinary computeValue also accepts reading in a Byte array, so I thought it might be confusing if the StringToHexBinary converter also allows reading in both a String or a Byte array.

Reply:
That is a little confusing, but is at least consistent with the StringTo{Date,Time,DateTime} objects. I'd vote for consistency and less duplication. Maybe create a bug to try to figure out why the StringTo* objects will also convert things that aren't Strings? Maybe that's correct behavior and they just need to be renamed to ConverToDateTime, for example.

Reply:
I've made the change and it should be pushed up now.